### DISCUSSION TITLE
Why is k=100 default for creating kNN aggregates?

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#1518] [Questions / Documentation]
Hi @rcorces and @jeffmgranja,

I am trying to gain more understanding and intuition regarding the creation of low overlapping aggregates or metacells in ArchR procedures such as `addPeak2GeneLinks()`. The default size of these aggregates or groups is 100 cells:

```
addPeak2GeneLinks(
  ArchRProj = NULL,
  reducedDims = "IterativeLSI",
  useMatrix = "GeneIntegrationMatrix",
  dimsToUse = 1:30,
  scaleDims = NULL,
  corCutOff = 0.75,
  cellsToUse = NULL,
  k = 100
)
```

Do you have any ideas on how the downstream results may be affected if k were increased or decreased?  Perhaps a lower k may lead to more spurious correlations, while a higher k would lead to more informative aggregates (albeit, with less power as there would be less aggregates to test the correlations on). Did you every benchmark this parameter to find out how it affects the number of significant P2Gs identified? 

Thank you for your help! I look forward to gaining a deeper understanding of this process.

**Comment by rcorces** [comment]
I do not believe that we ever benchmarked this parameter. But I also dont believe that benchmarking it is super informative for broad application. The specific value for k is largely dependent on your dataset size and the size of your clusters. I believe that higher k with small clusters would lead to cross-cluster inclusion in the low-overlapping cell aggregates. `k=100` is a pretty good starting point because if you cluster has really few cells (less than 200) then you arent going to be able to do much with it anyways. And aggregating less than 100 cells might not give you enough signal to be informative / avoid high amounts of noise.

**Reply by RegnerM2015** [reply]
Thank you for discussing! I appreciate your insights.
